libvast/src/view.cpp

```diff
@@ -53,6 +54,40 @@ bool operator<(pattern_view x, pattern_view y) noexcept {
   return x.string() < y.string();
 }
 
+bool is_equal(const data& x, const data_view& y) {
+  auto pred
+    = [](const auto& lhs, const auto& rhs) { return is_equal(lhs, rhs); };
+  auto f = detail::overload(
+    [&](const auto& lhs, const auto& rhs) {
+      using lhs_type = std::decay_t<decltype(lhs)>;
+      using rhs_type = std::decay_t<decltype(rhs)>;
+      if constexpr (std::is_same_v<view<lhs_type>, rhs_type>)
```

Comment:
Without decaying, I get a `const T&` as type, which is not compatible with `view<T>`. Hence the decaying.

Unfortunately `is_equality_comparable` doesn't work, because `data` has a non-explicit constructor. (See below for details.)

Reply:
I see. I suppose this is fine then.